### PR TITLE
fix: Permission Components #6199

### LINKED ISSUE #6199: Permission Components
- [x] add translation chunk for permission, add necessary translations
- [x] prepare details view
- [x] prepare form component - dynamic fields (missing backend data can be mocked and noted)

more: https://wiki.hybris.com/pages/viewpage.action?spaceKey=spar&title=Common+B2B+Commerce+Org+UX+-+Listings%2C+Assignments%2C+Viewing+and+Creating

### PR DESCRIPTION
**Pull request by architworld**
Closes #6199

### CODE REVIEW

File: projects/storefrontlib/src/cms-components/organization/permissions/permission-form/permission-form.component.ts [hunk 1]
```diff
@@ -74,6 +75,10 @@ export class PermissionFormComponent implements OnInit {
     threshold: ['', Validators.required],
   });
 
+  timeControl = this.form.get('periodRange');
```
Comment (Pucek9) [inline projects/storefrontlib/src/cms-components/organization/permissions/permission-form/permission-form.component.ts]: IMO `periodControl` is more precise

File: projects/storefrontlib/src/cms-components/organization/permissions/permission-form/permission-form.component.html [hunk 2]
```diff
@@ -96,40 +133,10 @@
...
-              ></ng-container>
-            </label>
-          </div>
-        </ng-container>
-      </div>
-
-      <div class="form-group">
+      <div
+        class="form-group"
+        *ngIf="selectedPermissionType !== 'B2BBudgetExceededPermission'"
```
Comment (Pucek9) [inline projects/storefrontlib/src/cms-components/organization/permissions/permission-form/permission-form.component.html]: I think, we have already info that fields are enabled. What do you think about usage our currencyControl.enabled flag?

The same for all of dynamic fields

File: projects/storefrontlib/src/cms-components/organization/permissions/permission-form/permission-form.component.ts [hunk 3]
```diff
@@ -33,6 +33,7 @@ export class PermissionFormComponent implements OnInit {
   businessUnits$: Observable<B2BUnitNode[]>;
   currencies$: Observable<Currency[]>;
   permissionTypes$: Observable<OrderApprovalPermissionType[]>;
+  selectedPermissionType: OrderApprovalPermissionType['code'];
```
Comment (Pucek9) [inline projects/storefrontlib/src/cms-components/organization/permissions/permission-form/permission-form.component.ts]: If we don't use enum here, the type of selectedPermissionType is just string.
But we needn't that field usage in ngModel and ngIf anymore, so you can remove that

File: projects/storefrontlib/src/cms-components/organization/permissions/permission-form/permission-form.component.ts (hunk 1 above)
Comment (architworld) [reply projects/storefrontlib/src/cms-components/organization/permissions/permission-form/permission-form.component.ts]: yes, I have changed that

File: projects/storefrontlib/src/cms-components/organization/permissions/permission-form/permission-form.component.html (hunk 2 above)
Comment (architworld) [reply projects/storefrontlib/src/cms-components/organization/permissions/permission-form/permission-form.component.html]: done

File: projects/storefrontlib/src/cms-components/organization/permissions/permission-form/permission-form.component.ts (hunk 3 above)
Comment (architworld) [reply projects/storefrontlib/src/cms-components/organization/permissions/permission-form/permission-form.component.ts]: done

**Pull request merged**